### PR TITLE
Session finalize + Response collection

### PR DESCRIPTION
**Pull request by shumatepf**
Add endpoint that:
- ends session
- fetches and returns response

### CODE REVIEW

File: qualtrix/client.py [hunk 1]
```diff
@@ -61,10 +85,77 @@ def result_export():
...
+
+    labels = response["result"]["labels"]
+
+    result = response["result"]
+
+    # Assign survey response status
+    # Qualtrics API returns poorly documented boolean as string - unsure if it returns anything else
+    try:
+        survey_answers["status"] = (
+            "Complete" if eval(labels["finished"]) else "Incomplete"
```
Comment (dzaslavskiy) [inline qualtrix/client.py]: This is seems like a fun way to do it but is it safe?

File: qualtrix/api.py [hunk 2]
```diff
@@ -2,33 +2,46 @@
...
 
     return client.result_export()
 
-@router.post("/response")
+@router.post("/response/{responseId}")
 async def test(responseId: str):
 
-    client.get_response(responseId)
+    response = client.get_response(responseId)
+    print(response)
```
Comment (dzaslavskiy) [inline qualtrix/api.py]: use logger

File: qualtrix/client.py [hunk 3]
```diff
@@ -61,10 +85,77 @@ def result_export():
...
+        )
+    except:
+        survey_answers["status"] = labels["finished"]
+
+    answer = {
```
Comment (dzaslavskiy) [inline qualtrix/client.py]: should have this logic only once.

**Review by dzaslavskiy — CHANGES_REQUESTED** [review]

File: qualtrix/client.py (hunk 1 above)
Comment (shumatepf) [reply qualtrix/client.py]: It's safe with the try

File: qualtrix/client.py (hunk 3 above)
Comment (shumatepf) [reply qualtrix/client.py]: Using a different value from the API response will help this. I was using a value that appeared to be either 'True' or 'False', but did not guarantee their values. There is a safer option that is either 1 or 0.

File: qualtrix/client.py (hunk 1 above)
Comment (shumatepf) [reply qualtrix/client.py]: Other comment for the try/except clause describes solution that will remediate this.

File: qualtrix/api.py (hunk 2 above)
Comment (dzaslavskiy) [reply qualtrix/api.py]: you did not make the changes?

File: qualtrix/client.py (hunk 3 above)
Comment (dzaslavskiy) [reply qualtrix/client.py]: Sorry I was unclear. The code to pull data from surveyresult is duplicated.

File: qualtrix/api.py (hunk 2 above)
Comment (shumatepf) [reply qualtrix/api.py]: Ah, removed the other prints but the one you commented on.

File: qualtrix/client.py (hunk 3 above)
Comment (shumatepf) [reply qualtrix/client.py]: Gotcha, put the duplicated code in a helper function

File: qualtrix/client.py [hunk 4]
```diff
@@ -47,24 +64,86 @@ def result_export():
...
+def finalize_session(session_id: str, response_id: str):
+    """
+    Business logic for ending session, pulling response, and posting to gdrive
+    """
+    if not delete_session(session_id):
+        raise error.QualtricsError("Not Found sessionId")
+
+    # The session deletion api attempts to delete a session, must poll for a response
+    response = ""
+    for i in range(5):
```
Comment (dzaslavskiy) [inline qualtrix/client.py]: move the 5 and 2 to named constants

**Review by dzaslavskiy — CHANGES_REQUESTED** [review]

File: qualtrix/client.py (hunk 4 above)
Comment (shumatepf) [reply qualtrix/client.py]: done

**Review by dzaslavskiy — APPROVED** [review]

**Pull request merged**